src/main/scala/viper/gobra/translator/implementations/components/OptionImpl.scala

```diff
@@ -0,0 +1,439 @@
+        vpr.EqCmp(
+          optType(expr, typeVar)(),
+          optTypeSome(typeVar)()
+        )()
+      )()
+    )(domainName = domainName)
+  }
+
+  /**
+    * {{{
+    * axiom {
+    *   forall o : Option[T] :: { optType(o) }
+    *     optType(o) == optTypeNone() && o == optNone() ||
+    *     optType(o) == optTypeSome() && exists e : T :: o == optSome(e)
+    * }
+    * }}}
+    */
+  private lazy val optType_existence_axiom : vpr.DomainAxiom = {
+    val oDecl = vpr.LocalVarDecl("o", vpr.DomainType(domainName, Map[vpr.TypeVar, vpr.Type]())(Seq(typeVar)))()
+    val eDecl = vpr.LocalVarDecl("e", typeVar)()
+    val expr = optType(oDecl.localVar, typeVar)()
+
+    vpr.AnonymousDomainAxiom(
+      vpr.Forall(
+        Seq(oDecl),
+        Seq(vpr.Trigger(Seq(expr))()),
+        vpr.Or(
+          vpr.And(
+            vpr.EqCmp(expr, optTypeNone(typeVar)())(),
+            vpr.EqCmp(oDecl.localVar, none(typeVar)())()
+          )(),
+          vpr.And(
+            vpr.EqCmp(expr, optTypeSome(typeVar)())(),
+            vpr.Exists(Seq(eDecl), Seq(), vpr.EqCmp(oDecl.localVar, some(eDecl.localVar)())())()
+          )()
+        )()
+      )()
+    )(domainName = domainName)
+  }
+
+  /**
+    * {{{
+    * axiom {
+    *   forall o : Option[T] :: { optIsNone(o) } optType(o) == optTypeNone() <==> optIsNone(o)
+    * }
+    * }}}
+    */
+  private lazy val optIsNone_type_axiom : vpr.DomainAxiom = {
+    val oDecl = vpr.LocalVarDecl("o", vpr.DomainType(domainName, Map[vpr.TypeVar, vpr.Type]())(Seq(typeVar)))()
+    val expr = isNone(oDecl.localVar, typeVar)()
+
+    vpr.AnonymousDomainAxiom(
+      vpr.Forall(
+        Seq(oDecl),
+        Seq(vpr.Trigger(Seq(expr))()),
+        vpr.EqCmp(
+          vpr.EqCmp(optType(oDecl.localVar, typeVar)(), optTypeNone(typeVar)())(),
+          expr
+        )()
+      )()
+    )(domainName = domainName)
+  }
+
+  /**
+    * {{{
+    * axiom {
+    *   forall o : Option[T] :: { optIsSome(o) } optType(o) == optTypeSome() <==> optIsSome(o)
+    * }
+    * }}}
+    */
+  private lazy val optIsSome_type_axiom : vpr.DomainAxiom = {
+    val oDecl = vpr.LocalVarDecl("o", vpr.DomainType(domainName, Map[vpr.TypeVar, vpr.Type]())(Seq(typeVar)))()
+    val expr = isSome(oDecl.localVar, typeVar)()
+
+    vpr.AnonymousDomainAxiom(
+      vpr.Forall(
+        Seq(oDecl),
+        Seq(vpr.Trigger(Seq(expr))()),
+        vpr.EqCmp(
+          vpr.EqCmp(optType(oDecl.localVar, typeVar)(), optTypeSome(typeVar)())(),
+          expr
+        )()
+      )()
+    )(domainName = domainName)
+  }
+
+  /**
+    * {{{
+    * domain Option[T] {
+    *
+    *   // Constructors
+    *
+    *   function optSome(e : T) : Option[T]
+    *   function optNone() : Option[T]
+    *
+    *   // Destructor
+    *
+    *   function optGet(o : Option[T]) : T
+    *
+    *   // Constructor Types
+    *
+    *   function optType(o : Option[T]) : Int
+    *   unique function optTypeNone() : Int
+    *   unique function optTypeSome() : Int
```

Comment:
I was not even aware of this viper feature. Why do you not just substitute values such as 0 and 1?